src/Illuminate/Support/Str.php

```diff
@@ -813,9 +797,9 @@ public static function replaceArray($search, $replace, $subject)
     /**
      * Replace the given value in the given string.
      *
-     * @param  string|string[]|Enumerable<array-key, string>  $search
-     * @param  string|string[]|Enumerable<array-key, string>  $replace
-     * @param  string|string[]|Enumerable<array-key, string>  $subject
+     * @param  string|array<string>|Enumerable<array-key, string>  $search
+     * @param  string|array<string>|Enumerable<array-key, string>  $replace
+     * @param  string|array<string>|Enumerable<array-key, string>  $subject
      * @return string
      */
     public static function replace($search, $replace, $subject)
```

Comment:
We could also make this work with `iterable`, but then we need to call `iterator_to_array`, which might not be as efficient? Not sure if that's true though.

Reply:
Made it work with `iterable` now. Did some benchmarks and tried to keep the same performance by not over-naively passing everything to `collect`.